From 7c2c210de6091fdf262e5a309a0f99718778adf7 Mon Sep 17 00:00:00 2001
From: Loren HUANG <b02279@freescale.com>
Date: Thu, 29 Aug 2013 11:18:18 +0800
Subject: [PATCH 0238/1072] ENGR00277201-1 3.10 kernel panic when running gpu
 stress test

git://git.freescale.com/imx/linux-2.6-imx.git imx_3.10.17_1.0.0_beta
commit 5c4556d839dee3c4194824a2cb9b72431a6595b5

The kernel panic is caused by gcc 4.8.1 build logic.
8044b000:       e24bd010        sub     sp, fp, #16
8044b004:       e1500004        cmp     r0, r4
8044b008:       b1a00004        movlt   r0, r4
8044b00c:       a51b0018        ldrge   r0, [fp, #-24]
8044b010:       e89da810        ldm     sp, {r4, fp, sp, pc}
With this logic, r0 may return 0xffffffff randomly in Cortex A9 processor.

Current workaround is enable ARM_UNWIND to avoid such disassemble logic.

Signed-off-by: Loren HUANG <b02279@freescale.com>
Signed-off-by: Hongbo Zhong <hongbo.zhong@windriver.com>
---
 arch/arm/configs/imx_v7_defconfig |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/arch/arm/configs/imx_v7_defconfig b/arch/arm/configs/imx_v7_defconfig
index f032daf..eed0a2e 100644
--- a/arch/arm/configs/imx_v7_defconfig
+++ b/arch/arm/configs/imx_v7_defconfig
@@ -281,7 +281,6 @@ CONFIG_MAGIC_SYSRQ=y
 # CONFIG_SCHED_DEBUG is not set
 # CONFIG_DEBUG_BUGVERBOSE is not set
 # CONFIG_FTRACE is not set
-# CONFIG_ARM_UNWIND is not set
 CONFIG_SECURITYFS=y
 # CONFIG_CRYPTO_ANSI_CPRNG is not set
 # CONFIG_CRYPTO_HW is not set
-- 
1.7.5.4

